lib/Unicode/UCD.pm

```diff
@@ -2533,6 +2533,8 @@ change these into digits, and then call C<num> on the result.
 
 sub num ($;$) {
     my ($string, $retlen_ref) = @_;
+    croak __PACKAGE__, "::num: second parameter must be a scalar reference"
+                        if defined $retlen_ref && ref $retlen_ref ne "SCALAR";
```

Comment:
If `$$retlen_ref` contains a reference this throws unnecessarily.  I'd expect something like:
```
    croak __PACKAGE__, "::num: second parameter must be a scalar reference"
                        if defined $retlen_ref && ref $retlen_ref ne "SCALAR" 
                                               && ref $retlen_ref ne "REF";
```